packages/documentation/copy/pt/reference/Decorators.md

```diff
@@ -0,0 +1,573 @@
+
+Assim, as etapas a seguir são executadas ao avaliar vários decoradores em uma única declaração no TypeScript:
+
+1. As expressões para cada decorador são avaliadas de cima para baixo.
+2. Os resultados são chamados como funções de baixo para cima.
+
+Se fôssemos usar [fábrica de decoradores](#fábrica-de-decoradores), podemos observar esta ordem de avaliação com o seguinte exemplo:
+
+```ts
+function f() {
+  console.log("f(): avaliada");
+  return function (
+    alvo,
+    chaveDePropriedade: string,
+    descritor: descritorDePropriedade
+  ) {
+    console.log("f(): chamada");
+  };
+}
+
+function g() {
+  console.log("g(): avaliada");
+  return function (
+    alvo,
+    chaveDePropriedade: string,
+    descritor: descritorDePropriedade
+  ) {
+    console.log("g(): chamada");
+  };
+}
+
+class C {
+  @f()
+  @g()
+  method() {}
+}
+```
+
+Que imprimiria esta saída no console:
+
+```shell
+f(): avaliada
+g(): avaliada
+g(): chamada
+f(): chamada
+```
+
+## Avaliação de Decoradores
+
+Há uma ordem bem definida para como os decoradores aplicados a várias declarações, dentro de uma classe, são aplicados:
+
+1. _Decoradores de Parâmetros_, seguido por _Mêtodo_, _Acessório_ ou _Decoradores de Propriedades_ são aplicados para cada membro da instância.
+2. _Decoradores de Parâmetros_, seguido por _Mêtodo_, _Acessório_ ou _Decoradores de Propriedades_ são aplicados para cada membro estático.
+3. _Decoradores de Parâmetros_ são aplicados para o construtor.
+4. _Decoradores de classe_ são aplicados para a classe.
+
+## Decoradores de Classes
+
+O _Decorador de Dlasses_ é declarado antes de uma declaração de classe.
+O decorador de classe é aplicado ao construtor da classe e pode ser usado para observar, modificar ou substituir uma definição de classe.
+Um decorador de classe não pode ser usado em um arquivo de declaração, ou em qualquer outro contexto de ambiente (como em uma classe `declare`).
+
+A expressão para o decorador de classe será chamada como uma função em tempo de execução, com o construtor da classe decorada como seu único argumento.
+
+Se o decorador da classe retornar um valor, ele substituirá a declaração da classe pela função construtora fornecida.
+
+> NOTA&nbsp; Se você decidir retornar uma nova função de construtor, deve tomar cuidado para manter o protótipo original.
+> A lógica que aplica decoradores em tempo de execução **não** fará isso por você.
+
+A seguir está um exemplo de um decorador de classe (`@selada`) aplicado a classe `Recepcionista`
+
+```ts
+@selada
+class Recepcionista {
+  recepcionista: string;
+  constructor(mensagem: string) {
+    this.cumprimento = mensagem;
+  }
+  cumprimentar() {
+    return "Olá, " + this.cumprimento;
+  }
+}
+```
+
+Podemos definir o decorador `@selado` usando a seguinte declaração de função:
+
+```ts
+function selado(constructor: Function) {
+  Object.selado(constructor);
+  Object.selado(constructor.prototype);
```

Comment:
```suggestion
  Object.seal(constructor);
  Object.seal(constructor.prototype);
```
These are native JS functions and should not be translated: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/seal